python/dex/api.py

```diff
@@ -82,34 +80,23 @@ def dex_func(name, *signature):
 destroyContext = dex_func('dexDestroyContext', HsContextPtr, None)
 forkContext    = dex_func('dexForkContext',    HsContextPtr, HsContextPtr)
 
-eval      = dex_func('dexEval',      HsContextPtr, ctypes.c_char_p, HsContextPtr)
+eval      = dex_func('dexEval',      HsContextPtr, ctypes.c_char_p, ctypes.c_int)
 lookup    = dex_func('dexLookup',    HsContextPtr, ctypes.c_char_p, HsAtomPtr)
 freshName = dex_func('dexFreshName', HsContextPtr, ctypes.c_char_p)
 
 print     = dex_func('dexPrint',     HsContextPtr, HsAtomPtr, ctypes.c_char_p)
 toCAtom   = dex_func('dexToCAtom',   HsAtomPtr,    CAtomPtr,  ctypes.c_int)
 fromCAtom = dex_func('dexFromCAtom', CAtomPtr,                HsAtomPtr)
 
-createJIT  = dex_func('dexCreateJIT',  HsJITPtr)
-destroyJIT = dex_func('dexDestroyJIT', HsJITPtr, None)
-compile    = dex_func('dexCompile',    HsJITPtr, ExportCC, HsContextPtr, HsAtomPtr, NativeFunction)
-unload     = dex_func('dexUnload',     HsJITPtr, NativeFunction, None)
+compile    = dex_func('dexCompile', HsContextPtr, ExportCC, HsAtomPtr, NativeFunction)
+unload     = dex_func('dexUnload',  HsContextPtr, NativeFunction, None)
 
-getFunctionSignature  = dex_func('dexGetFunctionSignature', HsJITPtr, NativeFunction, NativeFunctionSignaturePtr)
+getFunctionSignature  = dex_func('dexGetFunctionSignature', HsContextPtr, NativeFunction, NativeFunctionSignaturePtr)
 freeFunctionSignature = dex_func('dexFreeFunctionSignature', NativeFunctionSignaturePtr, None)
 
 xlaCpuTrampoline = lib.dexXLACPUTrampoline
 
 init()
-jit = createJIT()
-nofree = False
-@atexit.register
-def _teardown():
-  global nofree
-  destroyJIT(jit)
-  fini()
-  nofree = True  # Don't destruct any Haskell objects after the RTS has been shutdown
```

Comment:
Why is this gone? This is actually very important

Reply:
Fixed (after discussing with you offline)